exercises/concept/blorkemon-cards/.meta/Exemplar.elm

```diff
@@ -0,0 +1,79 @@
+module BlorkemonCards exposing
+    ( compareShinyPower
+    , expectedWinner
+    , isMorePowerful
+    , maxPower
+    , sortByCoolness
+    , sortByMonsterName
+    )
+
+
+type alias Shiny =
+    Bool
+
+
+type alias Monster =
+    String
+
+
+type alias Power =
+    Int
+
+
+type alias Card =
+    ( Monster, Power, Shiny )
+
+
+isMorePowerful : Card -> Card -> Bool
+isMorePowerful ( _, power1, _ ) ( _, power2, _ ) =
+    power1 > power2
+
+
+maxPower : Card -> Card -> Power
+maxPower ( _, power1, _ ) ( _, power2, _ ) =
+    max power1 power2
+
+
+sortByMonsterName : List Card -> List Card
+sortByMonsterName =
+    let
+        getName ( name, _, _ ) =
+            name
+    in
+    List.sortBy getName
+
+
+sortByCoolness : List Card -> List Card
+sortByCoolness =
+    let
+        coolness ( _, power, shiny ) =
```

Comment:
Is this using a tuple to define a hierachical sort? That is, sorting first by one property, and if those are equal then by another property.

This is something that doesn't seem to come out of the box in Elm, but people have made their own solutions, which I think are worth highlighting.

This is my one:
https://github.com/ceddlyburge/elm-league-tables-from-google-sheets/blob/master/src/Calculations/SortBy.elm

And in use at:
https://github.com/ceddlyburge/elm-league-tables-from-google-sheets/blob/master/src/Calculations/PlayersFromLeagueGames.elm

I think this is the more official package, although it is experimental
https://package.elm-lang.org/packages/rtfeldman/elm-sorter-experiment/latest/Sort

Reply:
> Is this using a tuple to define a hierachical sort? That is, sorting first by one property, and if those are equal then by another property.

Exactly. Although students might want to take a different route, with nested  `case compare x y of`.

Those are great links, I will use them.

Reply:
Hierarchical sort is a surprisingly difficult thing in Elm, and something we can usefully teach students, so I think it is worth explicitly calling it out and mentioning the various possibilities. We don't need them to write the code I don't think (certainly not of all the options), but being aware of the situation, and knowing to come back to the concept / exercise for some links if they come across the situation in real life seems like it would be useful ...